velox/exec/ExchangeClient.cpp

```diff
@@ -133,16 +130,15 @@ std::unique_ptr<SerializedPage> ExchangeClient::next(
 void ExchangeClient::request(const RequestSpec& requestSpec) {
   for (auto& source : requestSpec.sources) {
     auto future = source->request(requestSpec.maxBytes);
-    if (future.valid()) {
-      auto& exec = folly::QueuedImmediateExecutor::instance();
-      std::move(future)
-          .via(&exec)
-          .thenValue(
-              [&](auto&& /* unused */) { request(pickSourcesToRequest()); })
-          .thenError(
-              folly::tag_t<std::exception>{},
-              [&](const std::exception& e) { queue_->setError(e.what()); });
-    }
+    VELOX_CHECK(future.valid());
+    auto& exec = folly::QueuedImmediateExecutor::instance();
```

Comment:
Should we deprecate uses of QueuedImmediateExecutor? fbcode recommends against using it: https://fb.workplace.com/groups/bettereng/posts/1866733370141049

Reply:
I don't think we can because the alternative is to use coroutines which require C++20 and we are on C++17.

```
Stop using InlineExecutor, QueuedImmediateExecutor and null executors.
Prefer coroutines and avoid explicit executors entirely. Instead rely on coroutines to propagate executors transparently.
```